### PR TITLE
Metadata

### PR DESCRIPTION
**Pull request by hamstah**

### CODE REVIEW

**Comment by mikekelly** [comment]
need to remove the circle build

File: lib/medusa.ex [hunk 1]
```diff
@@ -32,6 +32,15 @@ defmodule Medusa do
 
   def start(_type, _args) do
     ensure_config_correct()
+    env = Application.get_env(:medusa, Medusa)
+    GenServer.start_link(
+      MedusaConfig, %{
+	adapter: env[:adapter],
+	message_validator: env[:message_validator]
+      },
+      [name: :config]
```
Comment (kalifs) [inline lib/medusa.ex]: name is global, so I should be more specific

File: lib/medusa.ex [hunk 2]
```diff
@@ -50,14 +59,34 @@ defmodule Medusa do
...
+  end
+
   def publish(event, payload, metadata \\ %{}) do
-    Medusa.Broker.publish(event, payload, metadata)
+    metadata = cond do
+      Map.has_key?(metadata, :id) -> metadata
+      true -> Map.put(metadata, :id, UUID.uuid4)
+    end
+
+    metadata = case Logger.metadata[:meta] do
```
Comment (kalifs) [inline lib/medusa.ex]: this is confusing

consider

```elixir
   metadata = Map.merge(metadata, Logger.metadata[:meta] || %{})
```

File: lib/medusa.ex [hunk 3]
```diff
@@ -32,6 +32,15 @@ defmodule Medusa do
 
   def start(_type, _args) do
     ensure_config_correct()
+    env = Application.get_env(:medusa, Medusa)
+    GenServer.start_link(
```
Comment (mikekelly) [inline lib/medusa.ex]: still don't understand what problem this is solving.

File: lib/medusa.ex [hunk 4]
```diff
@@ -50,14 +59,34 @@ defmodule Medusa do
     Medusa.Broker.new_route(route, function, opts)
   end
 
+  defp is_message_valid?(event, payload, metadata) do
```
Comment (mikekelly) [inline lib/medusa.ex]: might be wrong, but I think elixir style is to stick private functions at the bottom of the module?

File: lib/medusa.ex (hunk 1 above)
Comment (mikekelly) [inline lib/medusa.ex]: isnt' this usually wrapped in a `start_link` inside the module, and the name of the process is `__MODULE__`?

File: lib/medusa.ex (hunk 4 above)
Comment (hamstah) [inline lib/medusa.ex]: Fixed.

**Pull request merged**